apps/web/pages/api/auth/[...nextauth].tsx

```diff
@@ -215,29 +215,26 @@ export default NextAuth({
   callbacks: {
     async jwt({ token, user, account }) {
       const autoMergeIdentities = async () => {
-        if (!hostedCal) {
-          const existingUser = await prisma.user.findFirst({
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            where: { email: token.email! },
-          });
-
-          if (!existingUser) {
-            return token;
-          }
+        const existingUser = await prisma.user.findFirst({
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          where: { email: token.email! },
```

Comment:
Could we parse this with zod to ensure this is typesafe? Haven't tested but something like this should stop eslint/TS complaining here 

<img width="556" alt="CleanShot 2022-09-05 at 13 40 02@2x" src="https://user-images.githubusercontent.com/55134778/188451492-0887a1bd-0f5c-43b9-860a-340ede506933.png">

Reply:
That can be done in a follow up